File: Source/Aladdin.CAPI.Java/src/aladdin/capi/ITransportAgreement.java
package aladdin.capi;
import java.security.*; 
import java.io.*; 

///////////////////////////////////////////////////////////////////////////
// Алгоритм формирования общего ключа
///////////////////////////////////////////////////////////////////////////
public interface ITransportAgreement extends IAlgorithm
{
	// действия стороны-отправителя
	TransportAgreementData wrap(
        IPrivateKey senderPrivateKey, IPublicKey senderPublicKey, 
        IPublicKey[] recipientPublicKeys, IRand rand, ISecretKey key) 
        throws IOException, InvalidKeyException;
    
	// действия стороны-получателя
	ISecretKey unwrap(IPrivateKey recipientPrivateKey, 
        IPublicKey publicKey, byte[] random, 
        byte[] encryptedKey, SecretKeyFactory keyFactory) throws IOException;
}

File: Source/Aladdin.CAPI/CAPI/ITransportAgreement.cs
using System; 

namespace Aladdin.CAPI
{
    ///////////////////////////////////////////////////////////////////////////
	// Формирование общего ключа 
    ///////////////////////////////////////////////////////////////////////////
	public interface ITransportAgreement : IAlgorithm
	{
	    // действия стороны-отправителя
	    TransportAgreementData Wrap(IPrivateKey senderPrivateKey, 
            IPublicKey senderPublicKey, 
            IPublicKey[] recipientPublicKeys, IRand rand, ISecretKey key
        );
	    // действия стороны-получателя
	    ISecretKey Unwrap(IPrivateKey recipientPrivateKey, 
            IPublicKey publicKey, byte[] random, 
            byte[] encryptedKey, SecretKeyFactory keyFactory
        ); 
	}
}
